1112-i386-txt-Add-Intel-TXT-verification-routines.patch

```
@@ -0,0 +1,326 @@
++
++  /* TODO: Sort out how to do this verifier once the event log handling is in place
++   *
++   * return verify_evt_log((event_log_container_t *)(unsigned long)
++   *                       elog_elt->event_log_phys_addr);
++   */
++
++  return GRUB_ERR_NONE;
++}
++
++static grub_err_t
++verify_ext_data_elts(struct grub_txt_heap_ext_data_element *elts,
++                     grub_uint64_t elts_size)
++{
++  struct grub_txt_heap_ext_data_element *elt = elts;
++  grub_err_t err;
++
++  if ( elts_size < sizeof(*elt) )
++     return grub_error (GRUB_ERR_OUT_OF_RANGE,
++                 N_("TXT heap ext data elements too small"));
++
++  for ( ; ; )
++    {
++      if ( elts_size < elt->size || elt->size == 0 )
++        return grub_error (GRUB_ERR_OUT_OF_RANGE,
++                    N_("TXT heap invalid element size: type: %d, size: %d"),
++                    elt->type, elt->size);
++
++      switch ( elt->type )
++        {
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_END:
++            return GRUB_ERR_NONE;
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_BIOS_SPEC_VER:
++            err = verify_bios_spec_ver_elt (elt);
++            if ( err )
++              return err;
++            break;
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_ACM:
++            err = verify_acm_elt (elt);
++            if ( err )
++              return err;
++            break;
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_STM:
++            /* Nothing to check, platform specific */
++            break;
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_CUSTOM:
++            err = verify_custom_elt (elt);
++            if ( err )
++              return err;
++            break;
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_TPM_EVENT_LOG_PTR:
++            err = verify_evt_log_ptr_elt (elt);
++            if ( err )
++              return err;
++            break;
++          case GRUB_TXT_HEAP_EXTDATA_TYPE_MADT:
++            /* Copy of ACPI MADT, not validating */
++            break;
```

Is this validated by the ACM?

---

Hmm, out of these, only `GRUB_TXT_HEAP_EXTDATA_TYPE_BIOS_SPEC_VER`, `_ACM`, `_STM`, `_END` (obviously) and `_CUSTOM` are to be found in BIOS data section according to spec, I think I'll change the code to return error instead of warning for each type not mentioned above.

`_MADT` in particular is part of SINIT to MLE data, which is produced by SINIT ACM. It isn't available at this point.